src/utils/TokenUtils.ts

```diff
@@ -17,3 +19,46 @@ export const getL2TokenAddresses = (l1TokenAddress: string): { [chainId: number]
     return details.addresses[CHAIN_IDs.MAINNET] === l1TokenAddress;
   })?.addresses;
 };
+
+/**
+ * Returns the contract address for a given token symbol and chainId.
+ * @param symbol A case-insensitive token symbol.
+ * @param chainId The chainId to resolve the contract address for.
+ * @returns The contract address for the given token symbol and chainId, or undefined if the token symbol is not supported.
+ */
+export const resolveContractFromSymbol = (symbol: string, chainId: string): string | undefined => {
+  return Object.values(TOKEN_SYMBOLS_MAP).find((details) => {
+    return details.symbol.toLowerCase() === symbol.toLowerCase();
```

Comment:
Since `TOKEN_SYMBOLS_MAP` always stores in upper-case, how about just promoting the input `symbol` to upper-case instead?

Reply:
My only pushback is that it only stores uppercase so far. I know in the FE we account for USDC.e with the lowercase e. I'm not sure we have a guarantee that `TOKEN_SYMBOLS_MAP.[x].symbol` will always be lowercase

Reply:
Good call on USDC.e and potentially also the Base variant (USDbC or whatever).